src/stream/write/mod.rs

```diff
@@ -210,19 +210,14 @@ impl<'a, W: Write> Encoder<'a, W> {
     }
 
     /// Returns a wrapper around `self` that will finish the stream on drop.
-    ///
-    /// # Panic
-    ///
-    /// Panics on drop if an error happens when finishing the stream.
     pub fn auto_finish(self) -> AutoFinishEncoder<'a, W> {
-        self.on_finish(Box::new(|result| {
-            result.unwrap();
-        }))
+        self.on_finish(Box::new(|_result| {}))
```

Comment:
I think you can avoid that `Box::new` even if it probably allocates a zero-size object:

```suggestion
        AutoFinishEncoder {
            encoder: Some(self),
            on_finish: None,
        }
```